dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```
@@ -592,6 +593,34 @@ private static InetSocketAddress getSocksProxyAddress() {
         return InetSocketAddress.createUnresolved(hostAndPort.getHost(), hostAndPort.getPort());
     }
 
+    @Nullable
+    private static InetSocketAddress getSocksProxyAddress(ClientConfiguration clientConfiguration) {
+        // This is a roundabout way to extract proxy information. SOCKS proxies are configured at
+        // a different stage from HTTP proxies, so we must search for any socks proxy that the
+        // ProxySelector can provide. 'selector.select(null)' could work to validate no real work
+        // is being done, but I think the 127.0.0.1 lookup should be safer.
+        // Ideally we could pass along the original ServiceConfigurationBlock (if it exists) which
+        // provides a more precise description of this data.
+        try {
+            InetSocketAddress address = clientConfiguration.proxy().select(URI.create("127.0.0.1")).stream()
+                    .filter(proxy -> proxy.type() == Proxy.Type.SOCKS)
+                    .map(Proxy::address)
+                    .filter(InetSocketAddress.class::isInstance)
+                    .map(InetSocketAddress.class::cast)
+                    .findFirst()
+                    .orElseGet(ApacheHttpClientChannels::getSocksProxyAddress);
+            if (address != null) {
+                log.debug("Found SOCKS proxy address", UnsafeArg.of("address", address));
+            } else {
+                log.debug("No SOCKS proxy found");
+            }
+            return address;
+        } catch (RuntimeException e) {
```

This seems a little scary (as in might hide errors). Is this just to avoid breaking existing stuff in prod in case there is a problem with the socks detection?

---

Ya, this should log at ERROR rather than DEBUG :+1:

---

You are correct, I added the catch to avoid adding risk to the production case.